### PR TITLE
attempt numpy2 migration

### PR DESCRIPTION
**Pull request by greglandrum**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Following suggestions from here: https://github.com/conda-forge/rdkit-feedstock/pull/156#issuecomment-2218447961

### CODE REVIEW

**Comment by greglandrum** [comment]
@jaimergp @mcs07 @pstjohn I think this PR gets us set up to support numpy2.0 builds from conda-forge.
It's ready for review

**Comment by mcs07** [comment]
@conda-forge-admin please rerender

**Review by mcs07 — APPROVED** [review]
LGTM!

File: recipe/meta.yaml [hunk 1]
```diff
@@ -15,7 +15,7 @@ source:
     - pyproject.patch
 
 build:
-  number: 0
+  number: 1
```
Comment (jakirkham) [inline recipe/meta.yaml:18]: Minor suggestion to drop the trailing blank space

```suggestion
  number: 1
```

**Review by jakirkham — COMMENTED** [review]
Thanks all! 🙏

Had a minor suggestion. Otherwise looks good

**Comment by jaimergp** [comment]
Shall we wait for https://github.com/conda-forge/rdkit-feedstock/pull/158 before merging this one?

**Comment by greglandrum** [comment]
> Shall we wait for #158 before merging this one?

I think it would be useful to get a version out there that supports numpy2.0 and given that #158 is a bit one, it's hard for me to tell when it is going to be done.

**Review by jaimergp — APPROVED** [review]

**Pull request merged**